pg_search/tests/pg_regress/expected/issue_2844-rls.out

```diff
@@ -0,0 +1,382 @@
+CREATE OR REPLACE FUNCTION app.update_contact_tags()
+    RETURNS TRIGGER AS $$
+BEGIN
+    IF TG_OP = 'INSERT' THEN
+        WITH updated_contacts AS (
+            SELECT
+                contact_id,
+                jsonb_object_agg(
+                        tag_id,
+                        jsonb_build_object('created_at', created_at, 'created_by', created_by)
+                ) as new_tags
+            FROM new_table
+            GROUP BY contact_id
+        )
+        UPDATE app.contacts c
+        SET tags = tags || new_tags
+        FROM updated_contacts ut
+        WHERE c.id = ut.contact_id;
+    ELSIF TG_OP = 'DELETE' THEN
+        WITH updated_contacts AS (
+            SELECT
+                contact_id,
+                ARRAY_AGG(tag_id) AS tags_to_remove
+            FROM old_table
+            GROUP BY contact_id
+        )
+        UPDATE app.contacts c
+        SET tags = tags - tags_to_remove
+        FROM updated_contacts ut
+        WHERE c.id = ut.contact_id;
+    END IF;
+
+    RETURN NULL;
+END
+$$ LANGUAGE plpgsql SECURITY DEFINER;
+CREATE TRIGGER new_contact_tags_trigger
+    AFTER INSERT ON app.contact_tags
+    REFERENCING NEW TABLE AS new_table
+    FOR EACH STATEMENT
+EXECUTE FUNCTION app.update_contact_tags();
+CREATE TRIGGER del_contact_tags_trigger
+    AFTER DELETE ON app.contact_tags
+    REFERENCING OLD TABLE AS old_table
+    FOR EACH STATEMENT
+EXECUTE FUNCTION app.update_contact_tags();
+--
+-- add some contacts
+--
+insert into app.contacts (name, org_id)
+select
+    ('Contact ' || n::varchar) AS name,
+    ('tenant-' || (n % 25)::varchar) as org_id
+from generate_series(1, 750) n;
+--
+-- switch to the role with RLS rules
+--
+SET ROLE "app_user";
+SET app.user_id = 'ad_singh';
+SET app.org_id = 'tenant-1';
+SET app.device_id = 'adimac';
+SET app.has_full_contacts_access = 'true';
+EXPLAIN (FORMAT TEXT, COSTS OFF, TIMING OFF) UPDATE "app"."contacts" AS t
+SET "assignee" = 'ad_singh'
+WHERE t.id IN (SELECT id FROM "app".contacts LIMIT 1)
+RETURNING t.*;
+                           QUERY PLAN                            
+-----------------------------------------------------------------
+ Update on contacts t
+   ->  Nested Loop Semi Join
+         Join Filter: ((t.id)::text = ("ANY_subquery".id)::text)
+         ->  Custom Scan (ParadeDB Scan) on contacts t
+               Table: contacts
+               Index: contacts_search_idx
+               Exec Method: NormalScanExecState
+               Scores: false
+               Tantivy Query: {}
+         ->  Subquery Scan on "ANY_subquery"
+               ->  Limit
+                     ->  Custom Scan (ParadeDB Scan) on contacts
+                           Table: contacts
+                           Index: contacts_search_idx
+                           Exec Method: TopNScanExecState
+                           Scores: false
+                              Top N Limit: 1
+                           Tantivy Query: {}
+(18 rows)
```

Comment:
I noted in #2844 that I think this query plan is bogus.  I don't believe we should be submitting a path for scans fully unqualified table scans, with or without a limit.

I'm not addressing this in this PR, but this is why I included the explain, so that when we do address this, this test will show us we make a change.